includes/class-indieauth-client-discovery.php

```diff
@@ -9,7 +9,12 @@ class IndieAuth_Client_Discovery {
 
 	public function __construct( $client_id ) {
 		$this->client_id = $client_id;
-		$this->html      = self::parse( $client_id );
+
+		if ( defined( 'INDIEAUTH_UNIT_TESTS' ) ) {
+			return;
+		}
+
+		$this->html = self::parse( $client_id );
```

Comment:
Feels slightly odd to set the client_id property and then pass in the already passed in $client_id from the constructor, instead of perhaps relying on the newly set property.